src/test/java/com/github/fridujo/rabbitmq/mock/MockPolicyUseCaseTests.java

```diff
@@ -0,0 +1,584 @@
+
+                MockPolicy policyA = MockPolicy.builder()
+                    .name("dead letter exchange policy")
+                    .pattern("data")
+                    .definition(MockPolicy.DEAD_LETTER_EXCHANGE, "rejected-ex")
+                    .build();
+
+                connectionFactory.setPolicy(policyA);
+                assertThat(connectionFactory.listPolicies()).containsExactly(policyA);
+
+                configureRejectOnQueue(channel, "data");
+                configureMessageCapture(channel, "rejected", messagesFromPolicyA);
+                configureMessageCapture(channel, "rejected-2", messagesFromPolicyB);
+
+                channel.basicPublish("", "data", null, "bytes".getBytes());
+
+                waitForMessageDelivered(messagesFromPolicyA);
+
+                assertThat(messagesFromPolicyA.get(0)).isEqualTo("bytes".getBytes());
+                assertThat(messagesFromPolicyB).isEmpty();
+
+                messagesFromPolicyA.clear();
+
+                MockPolicy policyB = MockPolicy.builder()
+                    .name("higher priority dead letter exchange policy")
+                    .pattern("data")
+                    .definition(MockPolicy.DEAD_LETTER_EXCHANGE, "rejected-ex-2")
+                    .priority(1)
+                    .build();
+
+                connectionFactory.setPolicy(policyB);
+                assertThat(connectionFactory.listPolicies()).containsExactly(policyA, policyB);
+
+                channel.basicPublish("", "data", null, "bytes".getBytes());
+
+                waitForMessageDelivered(messagesFromPolicyB);
+
+                assertThat(messagesFromPolicyB.get(0)).isEqualTo("bytes".getBytes());
+                assertThat(messagesFromPolicyA).isEmpty();
+
+                messagesFromPolicyB.clear();
+
+
+                MockPolicy higherPriorityPolicyA = policyA.toBuilder().priority(2).build();
+                connectionFactory.setPolicy(higherPriorityPolicyA);
+                assertThat(connectionFactory.listPolicies()).containsExactly(higherPriorityPolicyA, policyB);
+
+
+                channel.basicPublish("", "data", null, "bytes".getBytes());
+
+                waitForMessageDelivered(messagesFromPolicyA);
+
+                assertThat(messagesFromPolicyA.get(0)).isEqualTo("bytes".getBytes());
+                assertThat(messagesFromPolicyB).isEmpty();
+
+                messagesFromPolicyA.clear();
+
+                connectionFactory.deletePolicy(policyA.getName());
+                assertThat(connectionFactory.listPolicies()).containsExactly(policyB);
+
+                channel.basicPublish("", "data", null, "bytes".getBytes());
+
+                waitForMessageDelivered(messagesFromPolicyB);
+
+                assertThat(messagesFromPolicyB.get(0)).isEqualTo("bytes".getBytes());
+                assertThat(messagesFromPolicyA).isEmpty();
+            }
+        }
+    }
+
+    @Test
+    public void deletingPolicyThatDoesNotExistThrowsNoExceptions() {
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/ledoyen/rabbitmq-mock/pullRequest?prid=5818045)